server/users.test.js

```diff
@@ -1,43 +1,387 @@
 const request = require('supertest')
     , {expect} = require('chai')
     , db = require('APP/db')
-    , app = require('./start')
+    , app = require('./start');
 
 /* global describe it before afterEach */
 
 describe('/api/users', () => {
-  before('Await database sync', () => db.didSync)
-  afterEach('Clear the tables', () => db.truncate({ cascade: true }))
 
-  describe('GET /:id', () =>
-    describe('when not logged in', () =>
-      it('fails with a 401 (Unauthorized)', () =>
+  const User = db.User;
+
+  before('Await database sync', () => db.didSync);
+  afterEach('Clear the tables', () => db.truncate({ cascade: true }));
+
+  // 'GET /'
+  describe('GET /', () => {
+
+    describe('gets all users', () => {
+
+      before('Dummy data', () => {
+        return User.create({
+          firstName: 'a',
+          lastName: 'b',
+          email: 'a@b.com',
+          // photo: default,
+          admin: false,
+          creditCard: 5105105105105100,
+          creditCardExpirationDate: '12/18',
+          creditCardCVV: 333,
+          houseNumber: 1234,
+          addressLine1: 'c',
+          city: 'd',
+          state: 'NY',
+      })
+        .then( () => {
+          return User.create({
+            firstName: 'f',
+            lastName: 'd',
+            email: 'e@b.com',
+            // photo: default,
+            admin: false,
+            creditCard: 5105105105105100,
+            creditCardExpirationDate: '12/18',
+            creditCardCVV: 333,
+            houseNumber: 1234,
+            addressLine1: 'r',
+            city: 'g',
+            state: 'NY',
+          });
+        })});
+      afterEach('Clear the tables', () => db.truncate({ cascade: true }));
+
+      it('returns all users in database', (done) => {
+        request(app)
+          .get(`/api/users`)
+          .expect(200)
+          .end( (err, res) => {
+            if(err) return done(err);
+            expect(res.body).to.be.instanceOf(Array);
+            expect(res.body).to.have.length(2);
+            done();
+          });
+      });
+    });
+
+    describe('does not get non-existent users', () => {
+      it('returns 404 if no users in database', (done) => {
         request(app)
-          .get(`/api/users/1`)
-          .expect(401)
-      )))
+          .get(`/api/users`)
```

Comment:
Shouldn't we test an id that doesn't exist instead of no id given? 

.get(`/api/users/100`)